test/runnable/istring.d

```diff
@@ -0,0 +1,25 @@
+/*
+PERMUTE_ARGS:
+*/
+import std.conv : text;
+void main()
+{
+    int a = 42;
+    assert("a is 42" == text(i"a is $(a)"));
+    assert("a + 23 is 65" == text(i"a + 23 is $(a + 23)"));
+    int b = 93;
+    assert("42 + 93 = 135" == text(i"$(a) + $(b) = $(a + b)"));
+
+    assert(928 == add(900, 28));
+}
+
+string funcCode(string attributes, string returnType, string name, string args, string body)
+{
+    return text(iq{
+    $(attributes) $(returnType) $(name)($(args))
+    {
+        $(body)
+    }
+    });
```

Comment:
Perhaps add tests with other string literal syntaxes as well.

Reply:
and prolly the suffixes. `i"foo $(bar)"w` should translate to `("foo "w, bar)` or something; just repeat the suffixes an the generated string literals inside.